include/aikido/control/ros/RosBarrettHandPositionCommandExecutor.hpp

```diff
@@ -0,0 +1,81 @@
+// Assume it has 4 positions
+#ifndef AIKIDO_CONTROL_ROS_ROSBARRETTHANDPOSITIONCOMMANDEXECUTOR_HPP_
+#define AIKIDO_CONTROL_ROS_ROSBARRETTHANDPOSITIONCOMMANDEXECUTOR_HPP_
+#include <chrono>
+#include <future>
+#include <mutex>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <sensor_msgs/JointState.h>
+#include <pr_control_msgs/SetPositionAction.h>
+#include <actionlib/client/action_client.h>
+#include <Eigen/Dense>
+
+namespace aikido{
+namespace control {
+namespace ros {
+
+/// This class sends commands for the hand position to the ROS server
+class RosBarrettHandPositionCommandExecutor
+{
+public:
+  /// Constructor
+  /// \param[in] node ROS node handle for action client.
+  /// \param[in] serverName Name of the server to send traejctory to.
+  /// \param[in] connectionTimeout Timeout for server connection.
+  /// \param[in] connectionPollingPeriod Polling period for server connection.
+  template <typename DurationA, typename DurationB>
+  RosBarrettHandPositionCommandExecutor(
+    ::ros::NodeHandle node,
+    const std::string& serverName,
+    const DurationA& connectionTimeout = std::chrono::milliseconds{1000},
+    const DurationB& connectionPollingPeriod = std::chrono::milliseconds{20}
+  );
```

Comment:
Nit: I don't think there is any value in templateing `DurationA` and `DurationB` here, since we immediately store these values in `std::chrono::milliseconds` member variables. To make this work, we would have to template the entire class on these two types - which seems like it is distinctly _not_ worth it.

I believe @jslee02, @gilwoolee, and I discussed this in a previous PR. Tagging them to see if they have anything to add.